### ISSUE TITLE
Claimable Balances Query slow on Horizon

### ISSUE THREAD

**Issue by jcx120** [stellar/go#4436]
Issue:

Claimable Balances query performing slow on SDF Horizon instance sporadically (50% of the time):

Query:
[https://horizon.stellar.org/claimable_balances/{balance_id}/transactions?limit=1&order=asc](https://horizon.stellar.org/claimable_balances/%7Bbalance_id%7D/transactions?limit=1&order=asc)

Issue:
This query is being used by a Customer to locate a transaction which created a claimable balance and retrieve its memo (since they can’t store a memo with claimable balance itself). While this call was not terribly fast, it got really bad this week, with the call taking approximately 8 seconds to either complete or error out with the following:

```
{

  "type": "https://stellar.org/horizon-errors/service_unavailable",

  "title": "Service Unavailable",

  "status": 503,

  "detail": "The request cannot be serviced at this time."

}
```

It works about 50% of the time with a 503 occurring the other 50%.

Sample URL:

https://horizon.stellar.org/claimable_balances/0000000028b32c721976af53fa0419eba129f3da70168f1e39822a25886375abad9599da/transactions?limit=1&order=asc

> Curious if the database optimization changes in Core 19 may have impacted this or if anyone has observed issues with this query. Also, is there an easier way to get original transaction which created claimable balance?

Note that this query returns pretty fast on an ingestion filtered [instance](https://horizon-c2e21c.prototypes.kube001.services.stellar-ops.com/claimable_balances/0000000028b32c721976af53fa0419eba129f3da70168f1e39822a25886375abad9599da/transactions?limit=1&order=asc).

**Comment by jcx120** [comment]
Could this be related (ie: performance degradation from accumulated unreaped CB history): https://github.com/stellar/go/issues/4396

**Comment by sreuland** [comment]
@jcx120 , this appears to be duplicate, reporting slow query response situation on cb API path as reported on #4455 , which had a PR merged for fix - https://github.com/stellar/go/pull/4477, so I think this one can be moved to closed/done same as #4455, I think it will be in next release after 2.19.0.
